design/dhcp-less.md

```diff
@@ -0,0 +1,391 @@
+<!--
+ This work is licensed under a Creative Commons Attribution 3.0
+ Unported License.
+
+ http://creativecommons.org/licenses/by/3.0/legalcode
+-->
+
+# DHCP-less
+
+Discuss options and outline a proposal to enable DHCP-less leveraging existing
+Ironic support for this functionality, without dependencies on downstream customizations.
+
+## Status
+
+implementable
+
+## Summary
+
+Metal<sup>3</sup> provides an IPAM controller which can be used to enable
+deployment with static-IPs instead of DHCP, however currently it is not
+possible to use this functionality in a fully DHCP-less environement without
+downstream customizations.
+
+This proposal outlines the outstanding issues, and potential solutions to
+enable an improved DHCP-less solution for Metal<sup>3</sup> users.
+
+## Motivation
+
+Infrastructure management via Metal<sup>3</sup> in DHCP-less environments
+is common, but today our upstream features only partially solve for this use-case.
+
+Since there are several groups in the community who require this functionality,
+it makes sense to collaborate and ensure we can use upstream components where
+possible and only depend on downstream customizations where absolutely required.
+
+### Goals
```

Comment:
The BMO changes are mentioned in the `Static network configuration (no IPAM)` user story and since that part is already solved we don't need further coverage of it here?

And the IPA agent dependencies are mentioned but not covered in detail since there are many potentially valid approaches and different tools that can consume the network_data.json - community consensus around the tooling would be nice to have, but it's not essential (and the tool choice may be coupled to the target OS) so I'm not trying to solve that here.

Perhaps I should remove the first paragraph and just leave the `Enable e2e integration of the CAPM3 IPAM components ...` part?

Reply:
Yes I think keeping just the last part would be clearer.

Reply:
I removed the first part, and also moved/renamed the proposal to make it clear it's only solving the IPAM/templating part